apis/core/test/domain/column-mapping/delete.test.ts

```diff
@@ -0,0 +1,127 @@
+      })
+
+    columnMapping = clownface({ dataset: $rdf.dataset() })
+      .node($rdf.namedNode('columnMapping'))
+      .addOut(rdf.type, cc.ColumnMapping)
+      .addOut(rdf.type, hydra.Resource)
+      .addOut(cc.sourceColumn, $rdf.namedNode('my-column'))
+      .addOut(cc.targetProperty, $rdf.namedNode('test'))
+      .addOut(cc.datatype, xsd.integer)
+      .addOut(cc.language, $rdf.literal('fr'))
+      .addOut(cc.defaultValue, $rdf.literal('test'))
+
+    const otherColumnMapping = clownface({ dataset: $rdf.dataset() })
+      .node($rdf.namedNode('otherColumnMapping'))
+      .addOut(rdf.type, cc.ColumnMapping)
+      .addOut(rdf.type, hydra.Resource)
+      .addOut(cc.sourceColumn, $rdf.namedNode('my-column2'))
+      .addOut(cc.targetProperty, $rdf.namedNode('other'))
+
+    const table = clownface({ dataset: $rdf.dataset() })
+      .namedNode('myTable')
+      .addOut(rdf.type, cc.Table)
+      .addOut(cc.csvMapping, csvMapping)
+      .addOut(cc.csvSource, csvSource)
+      .addOut(cc.columnMapping, columnMapping)
+      .addOut(cc.columnMapping, otherColumnMapping)
+
+    columnMappingObservation = clownface({ dataset: $rdf.dataset() })
+      .node($rdf.namedNode('columnMappingObservation'))
+      .addOut(rdf.type, cc.ColumnMapping)
+      .addOut(rdf.type, hydra.Resource)
+      .addOut(cc.sourceColumn, $rdf.namedNode('my-column'))
+      .addOut(cc.targetProperty, $rdf.namedNode('test'))
+      .addOut(cc.datatype, xsd.integer)
+
+    observationTable = clownface({ dataset: $rdf.dataset() })
+      .namedNode('myObservationTable')
+      .addOut(rdf.type, cc.Table)
+      .addOut(rdf.type, cc.ObservationTable)
+      .addOut(cc.csvMapping, csvMapping)
+      .addOut(cc.csvSource, csvSource)
+      .addOut(cc.columnMapping, columnMappingObservation)
+
+    dimensionMetadata = clownface({ dataset: $rdf.dataset() })
+      .namedNode('myDimensionMetadata')
+      .addOut(rdf.type, cc.DimensionMetadataCollection)
+      .addOut(schema.hasPart, $rdf.namedNode('myDimension'), dim => {
+        dim.addOut(schema.about, $rdf.namedNode('test'))
+      })
+
+    store = new TestResourceStore([
+      table,
+      observationTable,
+      csvMapping,
+      csvSource,
+      dimensionMetadata,
+      columnMapping,
+      otherColumnMapping,
+      columnMappingObservation,
+    ])
+
+    getDimensionMetaDataCollection = sinon.stub().resolves(dimensionMetadata.term.value)
+    dimensionMetadataQueries = { getDimensionMetaDataCollection }
+    getTableForColumnMapping = sinon.stub().resolves(table.term.value)
+    tableQueries = {
+      getLinkedTablesForSource,
+      getTablesForMapping,
+      getTableForColumnMapping,
+    }
+  })
+
+  it('deletes a column mapping', async () => {
+    const resource = $rdf.namedNode('columnMapping')
+    const columnMapping = deleteColumnMapping({ resource, store, dimensionMetadataQueries, tableQueries })
+
+    expect(columnMapping).to.be.fulfilled
```

Comment:
It would be nice to check that the resource, and potential related dimension, are actually deleted